src/runtime/TemporalObject.cpp

```diff
@@ -1094,6 +1104,86 @@ int TemporalCalendarObject::toISOWeekOfYear(ExecutionState& state, const int yea
     return weekNum;
 }
 
+Value TemporalCalendarObject::calendarDateAdd(ExecutionState& state, const Value& calendar, const Value& date, const Value& duration, const Value& options, Value dateAdd)
+{
+    ASSERT(calendar.isObject());
+    ASSERT(options.isObject() || options.isUndefined());
+
+    if (dateAdd.isUndefined()) {
+        dateAdd = Object::getMethod(state, calendar, state.context()->staticStrings().lazydateAdd());
```

Comment:
should this happen for undefined, or other non-callable codes?

Reply:
If `dateAdd` is not present (undefined) then this should happen.